integration/cluster.go

```diff
@@ -0,0 +1,224 @@
+		if !exists {
+			continue
+		}
+		if e := instance.Close(); e != nil {
+			err = multierr.Append(err, e)
+		}
+	}
+	if !obj.Preserve {
+		if obj.dir == "" || obj.dir == "/" {
+			panic("obj.dir is set to a dangerous path")
+		}
+		if err := os.RemoveAll(obj.dir); err != nil { // dangerous ;)
+			return errwrap.Wrapf(err, "can't remove instance dir")
+		}
+	}
+	return err
+}
+
+// RunLinear starts up each instance linearly, one at a time.
+func (obj *Cluster) RunLinear() error {
+	for i, h := range obj.Hostnames {
+		// build a list of earlier instances that have already run
+		seeds := []*Instance{}
+		for j := 0; j < i; j++ {
+			x := obj.instances[obj.Hostnames[j]]
+			seeds = append(seeds, x)
+		}
+
+		instance, exists := obj.instances[h]
+		if !exists {
+			return fmt.Errorf("instance `%s` not found", h)
+		}
+
+		if err := instance.Run(seeds); err != nil {
+			return errwrap.Wrapf(err, "trouble running instance `%s`", h)
+		}
+
+		// FIXME: consider removing this wait entirely
+		// wait for startup before continuing with the next one
+		ctx, cancel := context.WithTimeout(context.Background(), longTimeout*time.Second)
+		defer cancel()
+		if err := instance.Wait(ctx); err != nil { // wait to get a converged signal
+			return errwrap.Wrapf(err, "mgmt wait on instance `%s` failed", h) // timeout expired
+		}
+	}
+
+	return nil
+}
+
+// Kill the cluster immediately. This is a `kill -9` for if things get stuck.
+func (obj *Cluster) Kill() error {
+	var err error
+	// do this in reverse for fun
+	for i := len(obj.Hostnames) - 1; i >= 0; i-- {
+		h := obj.Hostnames[i]
+		instance, exists := obj.instances[h]
+		if !exists {
+			continue
+		}
+		if e := instance.Kill(); e != nil {
+			err = multierr.Append(err, e)
+		}
+	}
+	return err
+}
+
+// Quit sends a friendly shutdown request to the cluster. You can specify a
+// context if you'd like to exit earlier. If you trigger an early exit with the
+// context, then this will end up running a `kill -9` so it can return. Remember
+// to leave a longer timeout when using a context since this will have to call
+// quit on each member individually.
+func (obj *Cluster) Quit(ctx context.Context) error {
+	var err error
+	// do this in reverse for fun
+	for i := len(obj.Hostnames) - 1; i >= 0; i-- {
+		h := obj.Hostnames[i]
+		instance, exists := obj.instances[h]
+		if !exists {
+			continue
+		}
+		if e := instance.Quit(ctx); e != nil {
+			err = multierr.Append(err, e)
+		}
+	}
+	return err
+}
+
+// Wait until the first converged state is hit for each member in the cluster.
+// Remember to leave a longer timeout when using a context since this will have
+// to call wait on each member individually.
+func (obj *Cluster) Wait(ctx context.Context) error {
+	var err error
+	for _, h := range obj.Hostnames {
+		instance, exists := obj.instances[h]
+		if !exists {
+			continue
+		}
+		// TODO: do we want individual waits?
+		//ctx, cancel := context.WithTimeout(context.Background(), longTimeout*time.Second)
```

Comment:
Should this commented code be removed? If not you should explain with a comment why it is still in here, as to me it looks like dead code.